sway-lib-std/src/token.sw

```diff
@@ -84,33 +84,24 @@ pub fn force_transfer_to_contract(amount: u64, asset_id: ContractId, to: Contrac
 
 /// Transfer `amount` coins of type `asset_id` and send them to
 /// the address `to`.
-pub fn transfer_to_output(amount: u64, asset_id: ContractId, to: Address) {
+pub fn transfer_to_address(amount: u64, asset_id: ContractId, to: Address) {
     // maintain a manual index as we only have `while` loops in sway atm:
     let mut index = 0;
-    let mut output_index = 0;
-    let mut output_found = false;
 
     // If an output of type `OutputVariable` is found, check if its `amount` is
     // zero. As one cannot transfer zero coins to an output without a panic, a
     // variable output with a value of zero is by definition unused.
-    let outputs = output_count();
-    while index < outputs {
-        let type_of_output = output_type(index);
-        if let Output::Variable = type_of_output {
+    while index < output_count() {
```

Comment:
Does `output_count()` get called for each iteration of the loop here?
Or does the compiler somehow cache this value without the need to explicitly cache it in a temp var like `outputs` ?

Reply:
I recall comments about the compiler being smart enough to see if something is `read-only` or something. The implication was that if something does not change in some order of operations then repeating the value / call can have some pruning so that it's not called over and over. Not my area though

Reply:
Hmmm that's actually a very good question. I don't know. What I know is that we most definitely should *not* be doing that, at least in the future, once all of our loop optimizations are implemented. For that reason, I will revert this to how it was to be on the same side. Thanks!